src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -1022,10 +1022,23 @@ public class TraceStatus
             /// <value></value>
             public string? Hostname
             {
-                get => _status.Destination != IPAddress.Any.ToString()
-                    && _status.Destination != IPAddress.IPv6Any.ToString()
-                        ? _status.Destination
-                        : null;
+                get
+                {
+                    if (_status.Address?.ToString() == IPAddress.Any.ToString()
+                        || _status.Address?.ToString() == IPAddress.IPv6Any.ToString())
+                    {
+                        // There was no response to the ping (TimedOut).
+                        return null;
+                    }
+
+                    if (_status.Destination == string.Empty)
+                    {
+                        // There was a response, but the destination field is empty; use DisplayAddress.
+                        return _status.DisplayAddress;
+                    }
+
+                    return _status.Destination;
```

Comment:
Looking the code I guess there is a bug in ProcessTraceroute()
`if (ResolveDestination.IsPresent && routerName == string.Empty)` - we skip assigning the routerName

Reply:
Yeah. I left that in because I wanted to allow it to retry assigning the router name if resolve destination is present. I think I should split up that if statement, so if ResolveDestination isn't present at all it just immediately assigns the IP address.

Reply:
Fixed the ProcessTraceroute() issue by checking if `routerName` is `string.Empty` when creating the objects for output and instead assigning the `hopAddressString` as the `Destination` field instead of leaving it blank.